test/inference.jl

```diff
@@ -418,6 +418,10 @@ gpure(x::Irrational) = fpure(x)
 @test gpure() == gpure() == gpure()
 @test gpure(π) == gpure(π) == gpure(π)
 
+# Make sure @pure works for functions using the new syntax
+Base.@pure fpure2(x::T) where T = T
```

Comment:
preferable form is to write this as:
```julia
Base.@pure (fpure2(x::T) where T) = T
```
to avoid the garden-path visual ambiguity of seeing a trailing `T = T`